backend/src/src-licenses/src/main/java/com/siemens/sw360/licenses/db/LicenseDatabaseHandler.java

```diff
@@ -411,12 +416,13 @@ private License updateLicenseFromInputLicense(License license, License inputLice
         }
         license.setText(inputLicense.getText());
         license.setFullname(inputLicense.getFullname());
-        license.setId(inputLicense.getShortname());
+        license.setId(inputLicense.isSetId() ? inputLicense.getId() : inputLicense.getShortname());
```

Comment:
Didn't you want to fix this bug only once? Because there's PR #166 for this already.